### PR TITLE
Refactor (mini) 2 methods in ProviderMethods to make it easier/quicker to ever change them or debug them.

### PR DESCRIPTION
**Pull request by Martouta**
This couple of small changes makes it easier to change and easier to debug https://github.com/3scale/porta/pull/1939

### CODE REVIEW

File: app/models/account/provider_methods.rb [hunk 1]
```diff
@@ -338,13 +338,11 @@ def create_service(attrs)
   protected
 
   def create_first_service
-    return if @signup_mode
+    return if @signup_mode || has_bought_cinstance?
```
Comment (josemigallas) [inline app/models/account/provider_methods.rb:341]:
```suggestion
    return if @signup_mode || self.bought_cinstances.exists?
```

Comment (hallelujah) [inline app/models/account/provider_methods.rb:341]: That is not the same, in this case you will never run the ServiceCreator

File: app/models/account/provider_methods.rb [hunk 2]
```diff
@@ -338,13 +338,11 @@ def create_service(attrs)
...
 
-    unless self.bought_cinstances.exists?
-      application_plans = Account.master.accessible_services.default.application_plans
-      plan = application_plans.default || application_plans.first!
-      plan.create_contract_with!(self)
-    end
+    application_plans = Account.master.accessible_services.default.application_plans
+    plan = application_plans.default || application_plans.first!
+    plan.create_contract_with!(self)
```
Comment (hallelujah) [inline app/models/account/provider_methods.rb:345]: You can extract this in its new method

File: app/models/account/provider_methods.rb (hunk 1 above)
Comment (Martouta) [reply app/models/account/provider_methods.rb]: I know, that's why I left it as "WIP". sorry that I only spoke about it in private to another person 😞

File: app/models/account/provider_methods.rb (hunk 2 above)
Comment (Martouta) [reply app/models/account/provider_methods.rb]: better yes, done

**Pull request merged**